src/main/java/com/hivemq/cli/mqtt/AbstractMqttClientExecutor.java

```diff
@@ -200,10 +191,24 @@ public boolean isConnected(final @NotNull Subscribe subscriber) {
 
     public @NotNull MqttClient connect(final @NotNull Connect connect) {
 
-
         LoggingContext.put("identifier", "CLIENT " + connect.getIdentifier());
 
-        clientCache.setVerbose(connect.isVerbose());
+        if (isConnected(connect)) {
+            final MqttClient client = clientKeyToClientData.get(connect.getKey()).getClient();
+            switch (connect.getVersion()) {
+                case MQTT_5_0:
+                    ((Mqtt5Client) client).toBlocking().disconnect();
```

Comment:
can you use the mqtt5disconnect method here?

Reply:
Unfortunately this is not possible because a blocking api client is needed to wait for the disconnection of the client. Also, there would be a small architectural change to make.

Reply:
:(